packages/spanner/src/transaction.js

```diff
@@ -223,25 +228,26 @@ Transaction.prototype.begin = function(callback) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
```

Comment:
> do not support promises in runTransaction

I just don't understand how it works, and it doesn't match our pattern for the rest of our API in how we use callbacks vs promises. If this is how we supported promise usage, then I'd understand:

```js
database.runTransaction() // no callback? promise mode!
  .then(function([transaction]) {
    transaction.insert(...)
  })
  .catch(function([err]) {})
```

But `runTransaction` already take a function that gets executed *even in promise mode* to communicate if it failed. As I mentioned so far, the goal is maximum comprehension, intuitiveness, and predictability. As I think of this, `runTransaction` creates a sandbox. You can execute all the promises / non-promises you want in there, just like you can outside of it. But the walls of the sandbox are purposefully built so that there's predictability of where related behavior stays.

>  whether or not the expectation is to inform the user whether or not a Transaction was aborted while still retrying. I assumed that since we're retrying, we would only inform the user when we either got the final failure or a success (like any other API call retry strategy).

Yes, we would just "try to make it work" once, as far the user sees. Then, after all of our attempts, we communicate that it failed.